### PR TITLE
[update-price-p4] add reward calculation

### PR DESCRIPTION
**Pull request by ninjaahhh**
1. reward calc in `getRewardInfo`
2. reward calc when unstaking

should probably use safemath for uint256 to save some assertions.

### CODE REVIEW

File: contracts/UniversalV3Staker.sol [hunk 1]
```diff
@@ -273,23 +273,12 @@ contract UniversalV3Staker is IUniversalV3Staker, Multicall {
...
+        uint24 tickUpperShifted = uint24(deposit.tickUpper - TickMath.MIN_TICK + 1);
+        uint256 latestReward = _calculateReward(liquidity, tickLowerShifted, tickUpperShifted);
+        uint256 reward = latestReward - rewardDebt;
+        require(reward <= latestReward, 'UniswapV3Staker::getRewardInfo: overflow');
 
-        // if this overflows, e.g. after 2^32-1 full liquidity seconds have been claimed,
-        // reward rate will fall drastically so it's safe
-        incentive.totalSecondsClaimedX128 += secondsInsideX128;
         // reward is never greater than total reward unclaimed
         incentive.totalRewardUnclaimed -= reward;
```
Comment (qizhou) [inline contracts/UniversalV3Staker.sol:278]: I think totalRewardUnclaimed could be deleted.

File: contracts/UniversalV3Staker.sol [hunk 2]
```diff
@@ -428,26 +407,47 @@ contract UniversalV3Staker is IUniversalV3Staker, Multicall {
...
+    function _calculateReward(
+        uint128 liquidity,
+        uint24 tickLowerShifted,
+        uint24 tickUpperShifted
+    ) private view returns (uint256 reward) {
+        uint208 rshareLowerX64 = _cumulativeAccumulatedRewardsX64.get(_cfNbits, tickLowerShifted);
+        uint208 rshareUpperX64 = _cumulativeAccumulatedRewardsX64.get(_cfNbits, tickUpperShifted);
+        uint208 rshareX64 = rshareUpperX64 - rshareLowerX64;
+        require(rshareX64 <= rshareUpperX64, 'UniswapV3Staker::calculateReward: sub overflow');
+        uint256 rewardX64 = uint256(liquidity) * uint256(rshareX64);
```
Comment (qizhou) [inline contracts/UniversalV3Staker.sol]: Why not use SafeMath?

**Review by qizhou — APPROVED** [review]

File: contracts/UniversalV3Staker.sol (hunk 1 above)
Comment (ninjaahhh) [reply contracts/UniversalV3Staker.sol:278]: left here for book keeping reasons: `updatePrice` will check if the incentive exists or not and this field can be used.  we may clean up all unused variables in a later PR

File: contracts/UniversalV3Staker.sol (hunk 2 above)
Comment (ninjaahhh) [reply contracts/UniversalV3Staker.sol]: done

**Comment by ninjaahhh** [comment]
will fix `getRewardInfo` to include latest rewards in the next PR.

**Pull request merged**